### PR TITLE
add koa2-cors definition

### PR DESCRIPTION
**Pull request by xialeistudio**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not provide its own types, and you can not add them.
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, and `strictNullChecks` set to `true`.

### CODE REVIEW

File: types/koa2-cors/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
+// Type definitions for koa2-cors 2.0
+// Project: https://github.com/zadzbw/koa2-cors#readme
+// Definitions by: xialeistudio <https://github.com/xialeistudio>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 2.1
+
+import * as Koa from 'koa';
+declare namespace cors {
+    interface Options {
+        origin?: any;
```
Comment (plantain-00) [inline types/koa2-cors/index.d.ts]: The type can better be `string | (ctx: Koa.Context) => string` https://github.com/zadzbw/koa2-cors#origin

**Review by plantain-00 — CHANGES_REQUESTED** [review]

File: types/koa2-cors/index.d.ts (hunk 1 above)
Comment (xialeistudio) [reply types/koa2-cors/index.d.ts]: But it can be`boolean`,that make me confounded.
do you have any ideas?

Comment (plantain-00) [reply types/koa2-cors/index.d.ts]: The document says: `Configures the Access-Control-Allow-Origin CORS header. expects a string. Can also be set to a function, which takes the ctx as the first parameter.`
The source sode says this too: https://github.com/zadzbw/koa2-cors/blob/5c38ea409ab3cb6b73177012bda26ef81fd4d874/src/index.js#L5

Comment (xialeistudio) [reply types/koa2-cors/index.d.ts]: ok.I see,wait a moment

**Review by plantain-00 — APPROVED** [review]

File: types/koa2-cors/tsconfig.json [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    "compilerOptions": {
+        "module": "commonjs",
+        "lib": [
+            "es6"
+        ],
+        "noImplicitAny": true,
+        "noImplicitThis": true,
+        "strictNullChecks": true,
+        "baseUrl": "../",
+        "strictFunctionTypes": false,
```
Comment (andy-ms) [inline types/koa2-cors/tsconfig.json]: Why not?

Comment (xialeistudio) [reply types/koa2-cors/tsconfig.json]: I never use this option before,so ...

**Comment by andy-ms** [comment]
Thanks!

**Pull request merged**